src/textual/reactive.py

```diff
@@ -10,31 +10,125 @@
     Generic,
     Type,
     TypeVar,
-    Union,
 )
 
 import rich.repr
 
 from . import events
-from ._callback import count_parameters, invoke
-from ._types import MessageTarget
+from ._callback import count_parameters
 
 if TYPE_CHECKING:
-    from .app import App
-    from .widget import Widget
+    from .dom import DOMNode
 
-    Reactable = Union[Widget, App]
+    Reactable = DOMNode
```

Comment:
As mentioned, `MessagePump` can't really be a reactable, so the 2 lines that invoke `_initialize_reactive_object` and `_reset_reactable_object` have moved from there to `DOMNode`. `DOMNode` is the real base class for all reactables.